bundles/org.openhab.binding.monopriceaudio/src/main/java/org/openhab/binding/monopriceaudio/internal/handler/MonopriceAudioHandler.java

```diff
@@ -0,0 +1,749 @@
+        try {
+            connector.open();
+        } catch (MonopriceAudioException e) {
+            logger.debug("openConnection() failed: {}", e.getMessage());
+        }
+        logger.debug("openConnection(): {}", connector.isConnected() ? "connected" : "disconnected");
+        return connector.isConnected();
+    }
+
+    /**
+     * Close the connection with the MonopriceAudio device
+     */
+    private synchronized void closeConnection() {
+        if (connector.isConnected()) {
+            connector.close();
+            connector.removeEventListener(this);
+            logger.debug("closeConnection(): disconnected");
+        }
+    }
+
+    @Override
+    public void onNewMessageEvent(EventObject event) {
+        MonopriceAudioMessageEvent evt = (MonopriceAudioMessageEvent) event;
+
+        String key = evt.getKey();
+        String updateData = evt.getValue().trim();
+        if (!MonopriceAudioConnector.KEY_ERROR.equals(key)) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+        try {
+            switch (key) {
+                case MonopriceAudioConnector.KEY_ERROR:
+                    logger.debug("Reading feedback message failed");
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Reading thread ended");
+                    closeConnection();
+                    break;
+
+                case MonopriceAudioConnector.KEY_ZONE_UPDATE:
+                    String zoneId = updateData.substring(0, 2);
+
+                    if (MonopriceAudioZone.VALID_ZONES.contains(zoneId)) {
+                        MonopriceAudioZone targetZone = MonopriceAudioZone.valueOf(ZONE + zoneId);
+                        processZoneUpdate(targetZone, zoneDataMap.get(zoneId), updateData);
+                    } else {
+                        logger.warn("invalid event: {} for key: {}", evt.getValue(), key);
+                    }
+                    break;
+                default:
+                    logger.debug("onNewMessageEvent: unhandled key {}", key);
+                    break;
+            }
+        } catch (NumberFormatException e) {
+            logger.warn("Invalid value {} for key {}", updateData, key);
+        }
+    }
+
+    /**
+     * Schedule the reconnection job
+     */
+    private void scheduleReconnectJob() {
+        logger.debug("Schedule reconnect job");
+        cancelReconnectJob();
+        reconnectJob = scheduler.scheduleWithFixedDelay(() -> {
+            if (!connector.isConnected()) {
+                logger.debug("Trying to reconnect...");
+                closeConnection();
+                String error = null;
+                if (openConnection()) {
+                    synchronized (sequenceLock) {
+                        try {
+                            long prevUpdateTime = lastPollingUpdate;
+                            connector.queryZone(MonopriceAudioZone.ZONE1);
+                            Thread.sleep(150);
```

Comment:
A timeout of 150ms seem a bit short. Are you sure is is enough under all circumstances?

Reply:
Yes 150 should be more than enough here. Actually I will change it to use the constant.